### PR TITLE
Renamed _CShims to _FoundationCShims

### PR DESCRIPTION
**Pull request by iCharlesHu**
Rationale: _CShims will effectivly become semi-public in the toolchain. We add the Foundation prefix to make it less generic.

### CODE REVIEW

**Comment by iCharlesHu** [comment]
@swift-ci please test

**Review by jmschonfeld — APPROVED** [review]

**Comment by iCharlesHu** [comment]
Rebased

**Comment by iCharlesHu** [comment]
Rebased.

File: Sources/_FoundationCShims/include/module.modulemap [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module _FoundationCShims {
+    header "_FoundationCShims.h"
```
Comment (jmschonfeld) [inline Sources/_FoundationCShims/include/module.modulemap:2]: Does this need to be declared as an umbrella header?

**Pull request merged**